numpy/_core/tests/test_stringdtype.py

```diff
@@ -765,6 +765,12 @@ def test_multiply_reduce():
     assert res == val * np.prod(repeats)
 
 
+def test_multiply_two_string_raises():
+    arr = np.array(["hello", "world"])
```

Comment:
Oops, there is a mistake here: should have had `dtype=StringDType()` - since for regular `U`, one gets the no loop error already...